src/php/Evaluator/MethodNodeEvaluator/Contract/RouterTest.php

```diff
@@ -0,0 +1,294 @@
+<?php
+
+namespace Viamo\Floip\Evaluator\MethodNodeEvaluator\Contract;
+
+use Viamo\Floip\Evaluator\MethodNodeEvaluator\Contract\EvaluatesMethods;
+use Viamo\Floip\Evaluator\MethodNodeEvaluator\Contract\TestResult;
+
+interface RouterTest extends EvaluatesMethods
+{
+    const PUNCTUATION=',:;!?.-';
+
+    /**
+     * Tests whether all the words are contained in text.
+     * The words can be in any order and may appear more than once.
+     *
+     * @param string $text
+     * @param string $words
+     * @return TestResult
+     */
+    public function has_all_words($text, $words);
+
+    /**
+     * Tests whether any of the words are contained in the text
+     * Only one of the words needs to match and it may appear more than once.
+     *
+     * @param string $text
+     * @param string $words
+     * @return TestResult
+     */
+    public function has_any_word($text, $words);
+
+    /**
+     * Tests whether text starts with beginning.
+     * Both text values are trimmed of surrounding whitespace,
+     * but otherwise matching is strict without any tokenization.
+     *
+     * @param string $text
+     * @param string $beginning
+     * @return TestResult
+     */
+    public function has_beginning($text, $beginning);
+
+    /**
+     * Tests whether the category of a result on of the passed in categories
+     *
+     * @param array|Traversable|Node $result
+     * @param string ...$categories
+     * @return TestResult
+     */
+    public function has_category($result);
```

Comment:
where is the variadic param `string ...$categories`?

Reply:
I wish we could have that :cry: 
But some DNs are still running 5.5.9
So we must just _hint_ the variadic variable and use `func_get_args`

Reply:
uggghh.... 🤮